File: v1/backend/storage/repos/9f9d7705-da3e-45fb-b4fd-3b342766baf8/tuck-in-tales-frontend/src/models/prompt.ts
export interface Prompt {
  id: string;
  slug: string;
  version: number;
  is_active: boolean;
  name: string;
  description: string | null;
  system_prompt: string;
  user_prompt: string;
  provider: string;
  model: string;
  temperature: number | null;
  max_tokens: number | null;
  response_format: Record<string, unknown> | null;
  available_variables: string[];
  created_at: string;
  updated_at: string | null;
}

export interface PromptUpdate {
  name: string;
  description?: string | null;
  system_prompt: string;
  user_prompt: string;
  provider: string;
  model: string;
  temperature?: number | null;
  max_tokens?: number | null;
  response_format?: Record<string, unknown> | null;
  available_variables: string[];
}

export interface PromptTestRequest {
  system_prompt: string;
  user_prompt: string;
  provider: string;
  model: string;
  temperature?: number;
  max_tokens?: number | null;
}

export interface PromptTestResponse {
  response: string;
  provider: string;
  model: string;
  usage?: Record<string, unknown>;
}

File: v1/backend/storage/repos/9f9d7705-da3e-45fb-b4fd-3b342766baf8/tuck-in-tales-backend/src/models/prompt.py
from pydantic import BaseModel
from typing import Optional, List
from uuid import UUID
from datetime import datetime


class PromptBase(BaseModel):
    name: str
    description: Optional[str] = None
    system_prompt: str
    user_prompt: str
    provider: str = "openai"
    model: str = "gpt-4o-mini"
    temperature: Optional[float] = 0.7
    max_tokens: Optional[int] = None
    response_format: Optional[dict] = None
    available_variables: List[str] = []


class PromptUpdate(PromptBase):
    """Used when updating a prompt - creates a new version."""
    pass


class Prompt(PromptBase):
    id: UUID
    slug: str
    version: int
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class PromptTestRequest(BaseModel):
    system_prompt: str
    user_prompt: str
    provider: str
    model: str
    temperature: float = 0.7
    max_tokens: Optional[int] = None


class PromptTestResponse(BaseModel):
    response: str
    provider: str
    model: str
    usage: Optional[dict] = None
